packages/fullscreen/src/index.ts

```diff
@@ -0,0 +1,154 @@
+import { useState } from "react";
+
+type EventCallback = (this: Document, ev: any) => any;
+interface NormalizedFullscreenApi {
+  requestFullscreen: string;
+  exitFullscreen: string;
+  fullscreenElement: string;
+  fullscreenEnabled: string;
+  fullscreenchange: string;
+  fullscreenerror: string;
+}
+
+const getBrowserFunctions = (): NormalizedFullscreenApi => {
+  let val;
+
+  const fnMap = [
+    [
+      "requestFullscreen",
+      "exitFullscreen",
+      "fullscreenElement",
+      "fullscreenEnabled",
+      "fullscreenchange",
+      "fullscreenerror"
+    ],
+    // New WebKit
+    [
+      "webkitRequestFullscreen",
+      "webkitExitFullscreen",
+      "webkitFullscreenElement",
+      "webkitFullscreenEnabled",
+      "webkitfullscreenchange",
+      "webkitfullscreenerror"
+    ],
+    // Old WebKit
+    [
+      "webkitRequestFullScreen",
+      "webkitCancelFullScreen",
+      "webkitCurrentFullScreenElement",
+      "webkitCancelFullScreen",
+      "webkitfullscreenchange",
+      "webkitfullscreenerror"
+    ],
+    [
+      "mozRequestFullScreen",
+      "mozCancelFullScreen",
+      "mozFullScreenElement",
+      "mozFullScreenEnabled",
+      "mozfullscreenchange",
+      "mozfullscreenerror"
+    ],
+    [
+      "msRequestFullscreen",
+      "msExitFullscreen",
+      "msFullscreenElement",
+      "msFullscreenEnabled",
+      "MSFullscreenChange",
+      "MSFullscreenError"
+    ]
+  ];
+
+  let i = 0;
+  const l = fnMap.length;
+  const ret = {} as NormalizedFullscreenApi;
+
+  for (; i < l; i++) {
+    val = fnMap[i];
+    if (val && val[1] in document) {
+      for (i = 0; i < val.length; i++) {
+        ret[fnMap[0][i]] = val[i];
+      }
+      return ret;
+    }
+  }
+};
```

Comment:
Can we use forEach here instead of for loop? It would be could to use modern standards.

Reply:
I will change it, `for` loops usually have better performance to my knowledge but I agree that it would probably be more readable and since the loop is so small I am sure subscribers wont really feel it.

https://jsperf.com/for-vs-foreach/75